dali/imgcodec/parsers/pnm_test.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+#include "dali/imgcodec/image_source.h"
+#include "dali/imgcodec/parsers/pnm.h"
+#include "dali/test/dali_test.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali {
+namespace imgcodec {
+namespace test {
+
+class PnmParserTest : public ::testing::Test {
+ public:
+  PnmParserTest() {}
+
+  bool CanParse(std::vector<char> data) {
+    auto src = ImageSource::FromHostMem(data.data(), data.size());
+    return parser_.CanParse(&src);
+  }
+
+  PnmParser parser_;
+  const std::string pnm_directory_ = testing::dali_extra_path() + "/db/single/pnm/0/";
+};
+
+TEST_F(PnmParserTest, ValidPbm) {
+  auto img = ImageSource::FromFilename(pnm_directory_ + "cat-2184682_640.pbm");
+  EXPECT_TRUE(parser_.CanParse(&img));
+  TensorShape<> expected = {398, 640, 1};
+  EXPECT_EQ(parser_.GetInfo(&img).shape, expected);
+}
+
+TEST_F(PnmParserTest, ValidPgm) {
+  auto img = ImageSource::FromFilename(pnm_directory_ + "cat-1245673_640.pgm");
+  EXPECT_TRUE(parser_.CanParse(&img));
+  TensorShape<> expected = {423, 640, 1};
+  EXPECT_EQ(parser_.GetInfo(&img).shape, expected);
+}
+
+TEST_F(PnmParserTest, ValidPpm) {
+  auto img = ImageSource::FromFilename(pnm_directory_ + "cat-111793_640.ppm");
+  EXPECT_TRUE(parser_.CanParse(&img));
+  TensorShape<> expected = {426, 640, 3};
+  EXPECT_EQ(parser_.GetInfo(&img).shape, expected);
+}
+
+TEST_F(PnmParserTest, ValidPbmComment) {
+  const char data[] =
+      "P1\n"
+      "#This is an example bitmap of the letter \"J\"\n"
+      "6 10\n"
+      "0 0 0 0 1 0\n"
+      "0 0 0 0 1 0\n"
+      "0 0 0 0 1 0\n"
+      "0 0 0 0 1 0\n"
+      "0 0 0 0 1 0\n"
+      "0 0 0 0 1 0\n"
+      "1 0 0 0 1 0\n"
+      "0 1 1 1 0 0\n"
+      "0 0 0 0 0 0\n"
+      "0 0 0 0 0 0\n";
```

Comment:
const char data[] = R"_pnm_(P1
#This is an example bitmap of the letter "J"
6 10
0 0 0 0 1 0
0 0 0 0 1 0
0 0 0 0 1 0
0 0 0 0 1 0
0 0 0 0 1 0
0 0 0 0 1 0
1 0 0 0 1 0
0 1 1 1 0 0
0 0 0 0 0 0
0 0 0 0 0 0;
)_pnm_";
Something like this? You don't need escape sequences for new lines or quotes. It's a nitpick, though.